app/src/main/java/mozilla/lockbox/store/ItemDetailStore.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ *
+ */
+
+package mozilla.lockbox.store
+
+import io.reactivex.Observable
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.addTo
+import io.reactivex.subjects.BehaviorSubject
+import mozilla.lockbox.action.ItemDetailAction
+import mozilla.lockbox.action.RouteAction
+import mozilla.lockbox.extensions.filterByType
+import mozilla.lockbox.flux.Dispatcher
+
+class ItemDetailStore(
+    val dispatcher: Dispatcher = Dispatcher.shared
+) {
+    companion object {
+        val shared = ItemDetailStore()
+    }
+
+    private val compositeDisposable = CompositeDisposable()
+
+    internal val _passwordVisible = BehaviorSubject.createDefault(false)
+    val isPasswordVisible: Observable<Boolean> = _passwordVisible
+
+    init {
+        dispatcher.register
+            .filterByType(RouteAction::class.java)
+            .subscribe { action ->
+                when (action) {
+                    is RouteAction.ItemDetail -> _passwordVisible.onNext(false)
+                    is RouteAction.ItemList -> _passwordVisible.onNext(false)
+                }
+            }
+            .addTo(compositeDisposable)
```

Comment:
AFAIUI this is "each time you go from item list to item detail or back again, make the password invisible."

Is there a simpler way we can achieve this?

Reply:
nothing else was obvious.  Also note that this pattern is almost exactly what iOS does today.

Reply:
See my comment re: `ItemListPresenter` dispatching an `ItemDetailAction.View(id)` action, which itself dispatches the `RouteAction`.